src/VCEL.Core/Expression/Impl/TypeOperation.cs

```diff
@@ -0,0 +1,167 @@
+﻿namespace VCEL.Core.Expression.Impl
+{
+    public static class TypeOperation
+    {
+        public static bool EqualsMixedType(object left, object right)
+        {
+            switch (left)
+            {
+                case short shortLeft:
```

Comment:
We should order these so that the most likely types are compared first. I'd suggest the order: double, int, long, decimal, float, short, then all the unsigned.

Reply:
Updated the ordering.

Added a set of benchmarks to compare different cases.

## Results

|                  Method |       Mean |     Error |    StdDev | Ratio | RatioSD |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------------------ |-----------:|----------:|----------:|------:|--------:|-------:|------:|------:|----------:|
|              NullEquals |  0.8793 ns | 0.0362 ns | 0.0339 ns |  0.10 |    0.00 |      - |     - |     - |         - |
|       NullEqualsChecked |  0.8338 ns | 0.0042 ns | 0.0033 ns |  0.09 |    0.00 |      - |     - |     - |         - |
|             ShortEquals |  8.5792 ns | 0.1327 ns | 0.1241 ns |  0.96 |    0.02 | 0.0057 |     - |     - |      48 B |
|      ShortEqualsChecked |  8.0719 ns | 0.0441 ns | 0.0391 ns |  0.90 |    0.02 | 0.0057 |     - |     - |      48 B |
|            UshortEquals |  8.1752 ns | 0.1561 ns | 0.1460 ns |  0.91 |    0.02 | 0.0057 |     - |     - |      48 B |
|     UshortEqualsChecked |  8.3388 ns | 0.0815 ns | 0.0636 ns |  0.93 |    0.02 | 0.0057 |     - |     - |      48 B |
|               IntEquals |  8.9594 ns | 0.1977 ns | 0.1849 ns |  1.00 |    0.00 | 0.0057 |     - |     - |      48 B |
|        IntEqualsChecked |  8.6418 ns | 0.1617 ns | 0.1986 ns |  0.97 |    0.02 | 0.0057 |     - |     - |      48 B |
|              UintEquals |  8.7951 ns | 0.1566 ns | 0.1465 ns |  0.98 |    0.02 | 0.0057 |     - |     - |      48 B |
|       UintEqualsChecked |  8.4180 ns | 0.0896 ns | 0.0748 ns |  0.94 |    0.02 | 0.0057 |     - |     - |      48 B |
|              LongEquals |  8.4364 ns | 0.1938 ns | 0.2380 ns |  0.93 |    0.02 | 0.0057 |     - |     - |      48 B |
|       LongEqualsChecked |  8.4588 ns | 0.1476 ns | 0.1309 ns |  0.94 |    0.03 | 0.0057 |     - |     - |      48 B |
|             UlongEquals |  8.5500 ns | 0.1270 ns | 0.1126 ns |  0.95 |    0.02 | 0.0057 |     - |     - |      48 B |
|      UlongEqualsChecked |  8.4049 ns | 0.1755 ns | 0.1641 ns |  0.94 |    0.02 | 0.0057 |     - |     - |      48 B |
|            DoubleEquals |  9.0376 ns | 0.1947 ns | 0.1821 ns |  1.01 |    0.02 | 0.0057 |     - |     - |      48 B |
|     DoubleEqualsChecked |  8.9407 ns | 0.1923 ns | 0.1799 ns |  1.00 |    0.03 | 0.0057 |     - |     - |      48 B |
|           DecimalEquals | 14.0594 ns | 0.2932 ns | 0.2743 ns |  1.57 |    0.03 | 0.0076 |     - |     - |      64 B |
|    DecimalEqualsChecked | 13.6499 ns | 0.1115 ns | 0.0988 ns |  1.52 |    0.03 | 0.0076 |     - |     - |      64 B |
|            StringEquals |  1.0650 ns | 0.0323 ns | 0.0302 ns |  0.12 |    0.00 |      - |     - |     - |         - |
|     StringEqualsChecked |  1.0877 ns | 0.0214 ns | 0.0190 ns |  0.12 |    0.00 |      - |     - |     - |         - |
|        DecimalIntEquals | 15.5624 ns | 0.2549 ns | 0.2384 ns |  1.74 |    0.04 | 0.0067 |     - |     - |      56 B |
| DecimalIntEqualsChecked | 16.1343 ns | 0.1400 ns | 0.1242 ns |  1.80 |    0.03 | 0.0067 |     - |     - |      56 B |
|        IntDecimalEquals | 15.6798 ns | 0.1383 ns | 0.1155 ns |  1.75 |    0.05 | 0.0067 |     - |     - |      56 B |
| IntDecimalEqualsChecked | 15.0209 ns | 0.2164 ns | 0.2024 ns |  1.68 |    0.03 | 0.0067 |     - |     - |      56 B |
|        ULongFloatEquals | 14.3661 ns | 0.1670 ns | 0.1394 ns |  1.61 |    0.04 | 0.0057 |     - |     - |      48 B |
| ULongFloatEqualsChecked | 14.6473 ns | 0.1106 ns | 0.0980 ns |  1.64 |    0.03 | 0.0057 |     - |     - |      48 B |
|        FloatUlongEquals | 12.8779 ns | 0.1533 ns | 0.1359 ns |  1.44 |    0.04 | 0.0057 |     - |     - |      48 B |
| FloatUlongEqualsChecked | 13.3301 ns | 0.1391 ns | 0.1233 ns |  1.49 |    0.04 | 0.0057 |     - |     - |      48 B |